controllers/edgedeployment_controller.go

```diff
@@ -243,6 +293,29 @@ func (r *EdgeDeploymentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(r)
 }
 
+func (r *EdgeDeploymentReconciler) ExecuteConcurrent(funcs []concurrentFunc) []error {
```

Comment:
There is no need for this to be public here. You can create separate package responsible for concurrent execution and there it would make sense to have it public. It might even be useful for other packages.

Reply:
it was just temporary. I will add it to Utils.